sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,588 @@
+        )
+        self._criterion = (
+            self._config["criterion"]
+            if "criterion" in self._config.keys()
+            else self.criterion
+        )
+        # evaluate the sequence length passed by the user
+        self._seq_len = (
+            self._config["seq_len"]
+            if "seq_len" in self._config.keys()
+            else self.seq_len
+        )
+        if self._seq_len > self._moment_seq_len:
+            warnings.warn(
+                f"length of {self._seq_len} was found which is greater than 512. "
+                "The most recent 512"
+                " values will be used when fitting.",
+                stacklevel=2,
+            )
+            self._seq_len = 512
+        # in the case the config contains 'forecast_horizon', we'll set
+        # fh as that, otherwise we override it using the fh param
+        self._fh_config = (
+            self._config["forecast_horizon"]
+            if "forecast_horizon" in self._config.keys()
+            else self.fh
+        )
+        if self.device == "gpu" and torch.cuda.is_available():
+            self.device = "cuda"
+        else:
+            self.device = "cpu"
+
+        cur_epoch = 0
+        max_epoch = self.epochs
+        if fh is not None:
+            self._fh_input = max(fh.to_relative(self.cutoff))
+        self._fh = self._fh_input if fh is not None else self._fh_config
+        self._model = momentfm.MOMENTPipeline.from_pretrained(
+            self._pretrained_model_name_or_path,
+            model_kwargs={
+                "task_name": "forecasting",
+                "dropout": self._dropout,
+                "head_dropout": self._head_dropout,
+                "freeze_encoder": self._freeze_encoder,
+                "freeze_embedder": self._freeze_embedder,
+                "seq_len": 512,  # forced to be hard coded
+                "freeze_head": self._freeze_head,
+                "device": self._device,
+                "transformer_backbone": self._transformer_backbone,
+                "forecast_horizon": self._fh,
+            },
+        )
+        self._model.init()
+        self._y_cols = y.columns
+        self._y_shape = y.values.shape
+
+        # preparing the datasets
+        y_train, y_test = temporal_train_test_split(
+            y, train_size=1 - self.train_val_split, test_size=self.train_val_split
+        )
+
+        train_dataset = MomentPytorchDataset(
+            y=y_train,
+            fh=self._fh,
+            seq_len=self._seq_len,
+            device=self.device,
+        )
+
+        train_dataloader = DataLoader(
+            train_dataset, batch_size=self.batch_size, shuffle=True
+        )
+
+        val_dataset = MomentPytorchDataset(
+            y=y_test,
+            fh=self._fh,
+            seq_len=self._seq_len,
+            device=self.device,
+        )
+
+        val_dataloader = DataLoader(
+            val_dataset, batch_size=self.batch_size, shuffle=True
```

Comment:
```suggestion
            val_dataset, batch_size=len(val_dataset), shuffle=True
```

I think this adresses the batch processing thing I said before. I.e., all data would be processed at once.

Reply:
ah makes sense now! Thank you for clarifying